.github/workflows/test-agent.yml

```diff
@@ -0,0 +1,143 @@
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build daemon
+        run: >
+          docker run --rm --platform linux/${{matrix.arch}}
+          -v "${GITHUB_WORKSPACE}/php-agent":"/usr/local/src/newrelic-php-agent" 
+          $IMAGE_NAME:$IMAGE_TAG-${{ matrix.platform }}-$IMAGE_VERSION daemon
+      - name: Run daemon tests
+        run: >
+          docker run --rm --platform linux/${{matrix.arch}}
+          -v "${GITHUB_WORKSPACE}/php-agent":"/usr/local/src/newrelic-php-agent" 
+          $IMAGE_NAME:$IMAGE_TAG-${{ matrix.platform }}-$IMAGE_VERSION daemon_test
+  agent-unit-test:
+    runs-on: ubuntu-latest
+    env:
+      IMAGE_NAME: newrelic/nr-php-agent-builder
+      IMAGE_TAG: make-php
+      IMAGE_VERSION: v1
+    strategy:
+      matrix:
+        platform: [gnu, musl]
+        arch: [amd64, arm64]
+        php: ['5.5', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1', '8.2']
+        exclude:
+          - arch: arm64
+            php: '5.5'
+          - arch: arm64
+            php: '5.6'
+          - arch: arm64
+            php: '7.0'
+          - arch: arm64
+            php: '7.1'
+          - arch: arm64
+            php: '7.2'
+          - arch: arm64
+            php: '7.3'
+          - arch: arm64
+            php: '7.4'
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          path: php-agent
+      - name: Enable arm64 emulation
+        run: |
+          docker run --privileged --rm tonistiigi/binfmt --install arm64
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Determine if valgrind can be used
+        id: get-check-variant
+        run: |
+          if [[ ${{ matrix.platform }} = 'gnu' && ${{matrix.arch}} = 'amd64' ]]; then
+            echo "AXIOM_CHECK_VARIANT=valgrind" >> $GITHUB_OUTPUT
+          else
+            echo "AXIOM_CHECK_VARIANT=check" >> $GITHUB_OUTPUT
+          fi
+          if [[ ${{matrix.arch}} = 'arm64' ]]; then
+            echo "AGENT_CHECK_VARIANT=check" >> $GITHUB_OUTPUT
+          elif [[ ${{ matrix.platform }} = 'gnu' ]]; then
+            echo "AGENT_CHECK_VARIANT=valgrind" >> $GITHUB_OUTPUT
+          elif [[ ${{matrix.php}} = '5.5' || ${{matrix.php}} = '5.6' || ${{matrix.php}} = '7.0' || ${{matrix.php}} = '7.1' ]]; then
+            echo "AGENT_CHECK_VARIANT=check" >> $GITHUB_OUTPUT
+          else 
+            echo "AGENT_CHECK_VARIANT=valgrind" >> $GITHUB_OUTPUT
+          fi
+      - name: Build axiom
+        run: >
+          docker run --rm --platform linux/${{matrix.arch}}
+          -v "${GITHUB_WORKSPACE}/php-agent":"/usr/local/src/newrelic-php-agent" 
+          $IMAGE_NAME:$IMAGE_TAG-${{matrix.php}}-${{matrix.platform}}-$IMAGE_VERSION axiom 
```

Comment:
It seems like a good suggestion. Let me think about it.

Reply:
Option 1. Using the script it would look something like this:
```
ARCH=${{matrix.arch}} LIBC=${{matrix.platform}} PHP=${{matrix.php}} make.sh axiom
```
`make.sh` would have to check if all input params are provided and then hide away `docker run ...` command.


Option 2. Using reusable workflow it would look something like this:
```
  uses: ./.github/workflows/make-for-php.yml
    with:
      php: ${{matrix.php}}
      arch: ${{matrix.arch}}
      platform: ${{matrix.platform}}
      make-target: axiom
```
`make-for-php.yml` would hide away `docker run ...` command.

Does it look any better? I think Option 2. is more in line with GitHub Action's design.

Reply:
On the other hand I think this refactor could be done in its own PR. The workflow in its current shape achieves what is intended - it builds and unit tests daemon and agent on all supported platforms. Unless there's something functionally wrong with it, I'd like it to be merged and used, and then let's iterate over it.

Reply:
Agree with seeing the change in it's own PR.
Also really like the option 2 reusable workflow to reuse the matrix.* params and to keep the docker call in GHA.